tests/test_preload_script.py

```diff
@@ -95,3 +95,290 @@ async def test_addPreloadScript_logging(websocket, context_id):
     # Assert navigation is finished.
     result = await read_JSON_message(websocket)
     assert result == {"id": command_id, "result": ANY_DICT}
+
+
+# TODO: make offline.
+@pytest.mark.parametrize("nested_iframe", [
+    'about:blank',
+    'https://example.com/',
+])
+@pytest.mark.asyncio
+async def test_addPreloadScript_loadedInNewIframes(websocket, context_id,
+                                                   nested_iframe):
+    await subscribe(websocket, "log.entryAdded")
+
+    await execute_command(
+        websocket, {
+            "method": "script.addPreloadScript",
+            "params": {
+                "expression": "() => console.log('my preload script')",
+                "context": context_id,
+            }
+        })
+
+    command_id = await send_JSON_command(
+        websocket, {
+            "method": "browsingContext.navigate",
+            "params": {
+                "url": "data:text/html,<h2>test</h2>",
+                "wait": "complete",
+                "context": context_id
+            }
+        })
+
+    # Log event should happen before navigation.
+    result = await read_JSON_message(websocket)
+    assert result == AnyExtending({
+        "method": "log.entryAdded",
+        "params": {
+            "args": [{
+                "type": "string",
+                "value": "my preload script"
+            }]
+        }
+    })
+
+    # Assert navigation is finished.
+    result = await read_JSON_message(websocket)
+    assert result == {"id": command_id, "result": ANY_DICT}
+
+    # Create a new iframe within the same context.
+    result = await execute_command(
```

Comment:
Consider using `await send_JSON_command` / `await read_JSON_message` here to avoid timeout in case of preload script message was not sent. Same in other places. It's way better when test fails fast.